pkg/provisioner/reconciler/claim-reconciler/reconiler.go

```diff
@@ -262,10 +267,13 @@ func (r *ObjectBucketClaimReconciler) handleDeleteClaim(key client.ObjectKey) er
 	if err != nil || class == nil {
 		return fmt.Errorf("error getting storageclass from OB %q", ob.Name)
 	}
+
+	// use storage class to determine if this is a new or an old bucket
```

Comment:
a _dynamic provisioned_ flag should be stored in the OB. This is how we should decide greenfield vs. brownfield.